### PR TITLE
Added selectorsyncset generation

### PR DESCRIPTION
**Pull request by rafael-azevedo**
This pr adds the required steps to generate sync sets and the scaffolding to do so. It also adds the generation into the makefile.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,5 +1,20 @@
...
+$(error YAML_DIRECTORY is not set; check project.mk file)
+endif
+ifndef SELECTOR_SYNC_SET_TEMPLATE
+$(error SELECTOR_SYNC_SET_TEMPLATE is not set; check project.mk file)
+endif
+ifndef SELECTOR_SYNC_SET_DESTINATION
+$(error SELECTOR_SYNC_SET_DESTINATION is not set; check project.mk file)
+endif
+ifndef GIT_HASH
+$(error SELECTOR_SYNC_SET_DESTINATION is not set; check project.mk file)
```
Comment (jewzaam) [inline Makefile]:
```suggestion
$(error GIT_HASH is not set; check project.mk file)
```

**Review by jewzaam — CHANGES_REQUESTED** [review]

**Review by jewzaam — CHANGES_REQUESTED** [review]
There is no project.mk updates in this update.  Are you not defaulting vars like in other PRs?

**Review by jewzaam — APPROVED** [review]

**Review by jharrington22 — APPROVED** [review]
/lgtm

**Pull request merged**